sdk/include/platform/sunburst/platform-i2c.hh

```diff
@@ -0,0 +1,385 @@
+	  FifoControlTransmitReset = 1 << 8,
+	};
+
+	/**
+	 * Specification times (Table 10) in nanoseconds for each bus mode.
+	 */
+	static constexpr uint16_t SpcThigh[]  = {4000u, 600u, 260u};
+	static constexpr uint16_t SpcTlow[]   = {4700u, 1300u, 150u};
+	static constexpr uint16_t SpcThdSta[] = {4000u, 600u, 260u};
+	static constexpr uint16_t SpcTsuSta[] = {4700u, 600u, 260u};
+	static constexpr uint16_t SpcThdDat[] = {4000u, 1u, 1u};
+	static constexpr uint16_t SpcTsuDat[] = {500u, 100u, 50u};
+	static constexpr uint16_t SpcTBuf[]   = {4700u, 1300u, 500u};
+	static constexpr uint16_t SpcTsuSto[] = {4000u, 600u, 260u};
+
+	/// Reset all of the fifos
+	void reset_fifos() volatile
+	{
+		fifoCtrl = (FifoControlReceiveReset | FifoControlFormatReset |
+		            FifoControlAcquiredReset | FifoControlTransmitReset);
+	}
+
+	/// Configure the I2C block to be in host mode
+	void set_host_mode() volatile
+	{
+		control = ControlEnableHost;
+	}
+
+	/**
+	 * Set the I2C timing parameters appropriately for the given bit rate.
+	 * Distilled from:
+	 * https://github.com/lowRISC/opentitan/blob/9ddf276c64e2974ed8e528e8b2feb00b977861de/hw/ip/i2c/doc/programmers_guide.md
+	 */
+	void set_speed(const uint32_t SpeedKhz) volatile
+	{
+		// We must round up the system clock frequency to lengthen intervals.
+		constexpr uint32_t SysclkKhz = (I2cDevClockHz + 999) / 1000;
+		// We want to underestimate the clock period, to lengthen the timings.
+		uint32_t clkPeriod = (1000 * 1000) / SysclkKhz;
+
+		// Decide which bus mode this represents
+		uint32_t mode = (SpeedKhz > 100u) + (SpeedKhz > 400u);
+
+		// Calculation of timing parameters
+		uint16_t tHigh =
+		  round_up_divide(SpcThigh[mode], clkPeriod);              // Spec. min.
+		uint16_t tLow = round_up_divide(SpcTlow[mode], clkPeriod); // Spec. min.
+		uint16_t tFall =
+		  round_up_divide(20 * 3 / 5, clkPeriod); // Spec. min. 3.3V
+		uint16_t tRise = round_up_divide(120, clkPeriod);
+		// Setup and Hold times for Start
+		uint16_t thdSta = round_up_divide(SpcThdSta[mode], clkPeriod);
+		uint16_t tsuSta = round_up_divide(SpcTsuSta[mode], clkPeriod);
+		// Setup and Hold times for Data
+		uint16_t thdDat = round_up_divide(SpcThdDat[mode], clkPeriod);
+		uint16_t tsuDat = round_up_divide(SpcTsuDat[mode], clkPeriod);
+		uint16_t tBuf   = round_up_divide(SpcTBuf[mode], clkPeriod);
+		uint16_t tsuSto = round_up_divide(SpcTsuSto[mode], clkPeriod);
+
+		// Prevent counters underflowing
+		if (tLow < thdDat + 1u)
+			tLow = thdDat + 1u;
+		if (tBuf < tsuSta + 1u)
+			tBuf = tsuSta + 1u;
+
+		timing0 = (tLow << 16) | tHigh;
+		timing1 = (tFall << 16) | tRise;
+		timing2 = (thdSta << 16) | tsuSta;
+		timing3 = (thdDat << 16) | tsuDat;
+		timing4 = (tBuf << 16) | tsuSto;
+	}
+
+	void blocking_write_byte(const uint32_t Fmt) volatile
+	{
+		while (0 != (StatusFormatFull & status)) {}
+		formatData = Fmt;
+	}
+
+	/// Returns true when the format fifo is empty
+	[[nodiscard]] bool format_empty() volatile
```

Comment:
Predicates should normally include `is` or `are` in their names.  This one is particularly bad.  We have an explicit style rule to avoid words that are both verbs and nouns and, in this case, format and empty are *both* verbs and nouns and I don't know if this is a thing that formats an empty thing, empties a format, or some combination of the two.